torchdiffeq/_impl/misc.py

```diff
@@ -298,9 +302,40 @@ def _norm(tensor):
     # ~Backward compatibility
 
     # Add perturb argument to func.
-    func = _PerturbFunc(func)
+    wrapped_func = _PerturbFunc(func)
+
+    # Add callbacks to wrapped_func
+    callback_names = set()
+    for callback_name in _all_callback_names:
+        try:
+            callback = getattr(func, callback_name)
+        except AttributeError:
+            setattr(wrapped_func, callback_name, _null_callback)
+        else:
+            callback_names.add(callback_name)
+            # At the moment all callbacks have the arguments (t0, y0, dt).
+            # These will need adjusting on a per-callback basis if that changes in the future.
+            if is_tuple:
+                def callback(t0, y0, dt, _callback=callback):
+                    y0 = _flat_to_shape(y0, (), shapes)
+                    return _callback(t0, y0, dt)
+            if t_is_reversed:
+                def callback(t0, y0, dt, _callback=callback):
+                    return _callback(-t0, y0, dt)
+            setattr(wrapped_func, callback_name, callback)
+    for callback_name in _all_adjoint_callback_names:
+        try:
+            callback = getattr(func, callback_name)
+        except AttributeError:
+            pass
+        else:
+            setattr(wrapped_func, callback_name, callback)
+
+    invalid_callbacks = callback_names - SOLVERS[method].valid_callbacks()
```

Comment:
Can you also add a `valid_callbacks` class method into the scipy solver wrapper, so that this line runs for all solvers?

Reply:
Sure - I also spotted I missed the unit test from https://github.com/rtqichen/torchdiffeq/pull/134 so will add that in and double check.